### PR TITLE
[MIR] Avoid some code generation for stores of ZST

### LINKED ISSUE #30831: [MIR] Unecessary code for some ZST stores are generated
Code like the following one

```rust
use std::marker::PhantomData;

struct Zst { phantom: PhantomData<Zst> }

#[rustc_mir(graphviz="foo.gv")]
fn mir(){
    let x = Zst { phantom: PhantomData };
}
```

generates some unnecessary LLVM IR. Namely:

```llvm
bb0:                                              ; preds = %entry-block
  %0 = getelementptr inbounds %Zst, %Zst* %x, i32 0, i32 0
  store %"2.core::marker::PhantomData<Zst>" undef, %"2.core::marker::PhantomData<Zst>"* %0, align 1
  br label %bb1
```

We should skip zero-sized fields in [this code](https://github.com/rust-lang/rust/blob/1586005a1d15e7d26b6c46a33e7a9643de9a77ac/src/librustc_trans/trans/mir/rvalue.rs#L110).

### PR DESCRIPTION
**Pull request by nagisa**
Fixes #30831

r? @nikomatsakis

### CODE REVIEW

**Comment by nagisa** [comment]
Updated.

File: src/librustc_trans/trans/mir/operand.rs [hunk 1]
```diff
@@ -160,7 +160,6 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
         // stack.
 
         let o = self.trans_operand(bcx, operand);
-        self.store_operand(bcx, lldest, o);
```
Comment (nagisa) [inline src/librustc_trans/trans/mir/operand.rs]: Whoops!

**Comment by Manishearth** [comment]
This  or #30848 caused http://buildbot.rust-lang.org/builders/auto-win-msvc-64-opt/builds/2673/steps/test/logs/stdio

**Pull request merged**